File: src/main/java/skyblockclient/features/AntiBlind.java
package skyblockclient.features;

import net.minecraftforge.client.event.EntityViewRenderEvent;
import net.minecraftforge.client.event.RenderGameOverlayEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
import skyblockclient.SkyblockClient;
import skyblockclient.utils.SkyblockCheck;

public class AntiBlind {

    @SubscribeEvent
    public void renderFog(EntityViewRenderEvent.FogDensity event) {
        if (!SkyblockCheck.inSkyblock || !SkyblockClient.config.antiBlind) return;
        event.density = 0f;
        event.setCanceled(true);
    }

    @SubscribeEvent
    public void renderOverlay(RenderGameOverlayEvent event) {
        if (!SkyblockCheck.inSkyblock || !SkyblockClient.config.antiPortal) return;
        if (event.type == RenderGameOverlayEvent.ElementType.PORTAL) {
            event.setCanceled(true);
        }
    }

}

File: src/main/kotlin/skyblockclient/features/AntiBlind.kt
package skyblockclient.features

import net.minecraft.client.renderer.GlStateManager
import net.minecraftforge.client.event.EntityViewRenderEvent.FogDensity
import net.minecraftforge.client.event.RenderGameOverlayEvent
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
import skyblockclient.SkyblockClient.Companion.config
import skyblockclient.SkyblockClient.Companion.inSkyblock

class AntiBlind {

    @SubscribeEvent
    fun onRenderFog(event: FogDensity) {
        if (!config.antiBlind || !inSkyblock) return
        event.density = 0f
        GlStateManager.setFogStart(998f)
        GlStateManager.setFogEnd(999f)
        event.isCanceled = true
    }

    @SubscribeEvent
    fun onOverlay(event: RenderGameOverlayEvent.Pre) {
        if (!config.antiPortal || !inSkyblock) return
        if (event.type == RenderGameOverlayEvent.ElementType.PORTAL) {
            event.isCanceled = true
        }
    }
}
